torch/_subclasses/fake_tensor.py

```diff
@@ -333,8 +333,31 @@ def to_copy(fake_mode, func, *args, **kwargs):
 
     input_device = new_kwargs.pop("device", None)
     out_device = input_device if input_device else new_kwargs["input"].device
+    import pdb; pdb.set_trace()
+    """
+    Assumption: the reason for this no_dispatch is that we already recorded to_copy in our proxy graph,
+    and now we're doing the dirty work of actually running the real to_copy on our fake tensors to get
+    new 'real' (fake) outputs on the new dtype.  Hence, no_dispatch ensures no more tracing.
+        - note: it's superfluous in this case, as we are already in an equivalent no_dispatch region
+                inside proxy tensor and there are no modes active now
+
+        - other note: no_dispatch _shouldn't_ (afaiu) result in disabling meta key, which is kind of what
+          i'm suspecting is happening
+    """
     with no_dispatch():
-        input = new_kwargs.pop("input").to("meta")
+
+        """
+        Question 1- why is this .to("meta") here? shouldn't our input already be a faketensor?
```

Comment:
@eellison any insight on these questions?

Reply:
hmm.  i did not try to dump the dispatch keys of the tensor before/after .to("meta").  it strikes me as fishy that we `_like` constructors could produce a meta tensor that does not have meta dispatch key.  Isn't that an important invariant to maintain, meta tensor always has meta key?

If the meta tensor is indeed missing the meta key, I wonder if calling .to("meta") is a viable solution.  In my debugging, I saw the .to("meta") call trigger the very same infinite loop that I was trying to debug.  (to_copy decomp is entered, prim::device_put is called, it calls a.to("meta"), which again hits the to_copy decomp).

at least there is a way to force the meta key via TLS...  I would say this is my thought process:

1) if we determine that it is OK for some reason that meta tensors sometimes do not carry the meta key, then i think using the in_kernel_invocation_manager thing makes sense here as it reliably sets the meta key

2) if (1) is fishy, then maybe we shouldn't need the manager here at all, but we should figure out how to construct our meta tensors correctly in the first place

Reply:
Agreed that using `in_kernel_invocation_manager` makes sense here. 

See: https://github.com/pytorch/pytorch/pull/85337#pullrequestreview-1114091182/ 

Note that a tensor having a dispatch key is not always the same thing as `in_kernel_invocation_manager` - `in_kernel_invocation_manager` will set which kernel gets hit but is not actually adding `meta` to the dispatch keys of the input tensor physically. 

In any rate, I think adding the manager here makes sense.